certification/pyxis/builder_test.go

```diff
@@ -39,4 +43,218 @@ var _ = Describe("Pyxis Builder tests", func() {
 			Expect(int64(len(fileBytes))).To(Equal(info.Size()))
 		})
 	})
+
+	Context("When reading and storing files from disk", func() {
+		f := filepath.Join(tmpdir, "does-not-exist")
+		It("should fail when the file does not exist", func() {
+			err := readAndUnmarshal(f, nil)
+			Expect(err).To(HaveOccurred())
+		})
+	})
+
+	Context("When preparing a new input builder", func() {
+		Context("with a nil CertProject value", func() {
+			It("should return an error", func() {
+				_, err := NewCertificationInput(nil)
+				Expect(err).To(HaveOccurred())
+			})
+		})
+
+		Context("with a non-nil CertProject value as input", func() {
+			p := &CertProject{
+				ID:                  "myID",
+				CertificationStatus: "Pending",
```

Comment:
Is this a valid status? Do we care?

Reply:
We don't validate the status value within preflight, so I would say for the moment that we do not care.

Reply:
I believe that `Started`, `In Progress` and `Published` are the only valid statuses.